src/execution_plan/optimizations/optimize_cartesian_product.c

```diff
@@ -99,19 +108,21 @@ static void _optimize_cartesian_product(ExecutionPlan *plan, OpBase *cp) {
 			/* The optimization has depleted all of the cartesian product children, merged them and replaced the
 			 * cartesian product with the filter op.
 			 * Since the original cartesian product is no longer a valid operation, and there might be
-			 * additional filters which are applicable to re position after the optimization is done,
+			 * additional filters which are applicable to reposition after the optimization is done,
 			 * the following code tries to propagate up the remaining filters, and finish the loop. */
 			i++;
 			for(; i < filter_count; i++) {
-				ExecutionPlan_RePositionFilterOp(plan, (OpBase *)filter_op, NULL, (OpBase *)filters[i].filter);
+				ExecutionPlan_RePositionFilterOp(plan, (OpBase *)filter_op, NULL,
+												 (OpBase *)filter_ctx_arr[i].filter);
 			}
```

Comment:
Should we break out of the main loop `for(uint i = 0; i < filter_count; i ++)` at this point?

Reply:
no
we might be able to reposition additional filters
assume that every filter is with two entities, we have 3 branches cp for (a, b, c) nodes
and 3 filters:
1. a.x = b.x
2. b.x = c.x
3. a.y = b.y

the first will merge branches a and b to m1
the second will merge m1 and c to m2
the third can be repositioned below m1